packages/iframe-coordinator/src/messages/PromptOnLeave.ts

```diff
@@ -0,0 +1,30 @@
+import { boolean, constant, Decoder, object } from 'decoders';
+import { labeledDecoder, LabeledMsg } from './LabeledMsg';
+
+/** A prompt on leave notification to be displayed
+ * by the host application
+ */
+export interface PromptOnLeave {
+  /** A payload of type boolean */
+  payload: boolean;
```

Comment:
I would call this `shouldPrompt` or something else descriptive and update the doc comment to indicate that the host app will ask the user for confirmation before leaving the current page if it has received a message with the field set to true. 

I know I didn't capture this in the ticket, but after some research on the future of navigation APIs and what might be possible for us in the app, I'd also add an optional `message?: string` field documented as the message to prompt the user with (if possible).